osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/infra/jms/protocol/inbound/messageprocessors/EventNotificationMessageProcessor.java

```diff
@@ -56,15 +56,11 @@ public void processMessage(final ObjectMessage message) throws JMSException {
                 this.eventNotificationMessageService.handleEvent(metadata.getDeviceIdentification(), eventNotification);
 
             } else if (dataObject instanceof List) {
-
-                @SuppressWarnings("unchecked") final List<EventNotificationDto> eventNotifications =
-                        (List<EventNotificationDto>) dataObject;
-                this.eventNotificationMessageService.handleEvents(metadata.getDeviceIdentification(),
-                        eventNotifications);
+                throw new RuntimeException("a message of type List is not accepted");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00112)